### DISCUSSION TITLE
Is there a way to get the content-id for inline attachments on incoming mail?

### DISCUSSION THREAD

**Discussion by lindleym** [postalserver/postal#1963] [Help with using Postal]
Hello,

We've been testing a Postal implementation with fairly good results, but we are running into some occasional issues with handling incoming mail with inline attachments. The attachments array doesn't seem to include the content-id of the inline attachments. This makes accurately matching attachments up with the corresponding cid: sources in the html bodies impossible.
I've managed to work around some of these issues with varying degrees of sucess, and the accuracy of those workarounds is hard to determine.

Yesterday we received an incoming mail with only one attachment, but the message had two cids in the html source. It was a forwarded/resent message where the original attachment wasn't included, but the source still had the original cid. The second cid without a second attachment is tripping up my workarounds. I could work around this as well, but each workaround skews the accuracy of matching attahments to cids. Having the actual content-id in the attachments array of the JSON packet would be the ideal  solution to this problem. Is there any way to get that from the current incoming mail json packets?

**Comment by willpower232** [comment]
this is quite complicated so I can understand it not working properly at the moment, I wonder if the database already keeps the whole raw message or the parts are separated on being received so it would be harder to resolve, are you able to confirm what does and doesn't exist in the database?

**Reply by lindleym** [reply]
We were able to find the whole raw message in the raw-xxx-xx-xx table indicated in the messages table for the message in question, and verify that the content-id is in that raw data, so that is promising. What we don't know is how to get that extra data into the json packet sent to the endpoint for incoming messages. I found a couple of places where I think a check for a content-id would go, but I'm not failiar enough with Postal's codebase, or Ruby in general, to know exactly how to add this in without breaking things.

- https://github.com/postalserver/postal/blob/2e7aaaba5a9a37f32bdff9a9fde0379834b2a4db/lib/postal/http_sender.rb#L100-L103
- https://github.com/postalserver/postal/blob/2e7aaaba5a9a37f32bdff9a9fde0379834b2a4db/app/models/incoming_message_prototype.rb#L39-L41
